### PR TITLE
SC-5989 - language switcher base component which can set the default la…

### PR DESCRIPTION
**Pull request by mwolowyk-cap**
# Description

language switcher DUMMY component which can set the default language for school and user and send the request to the API

## Links to Tickets or other pull requests
https://ticketsystem.schul-cloud.org/browse/SC-5987

## Screenshots of UI changes

image

## Approval for review
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: helpers/i18n.js [hunk 1]
```diff
@@ -26,27 +31,46 @@ i18next
...
+	}
+
+	// get language by query
+	const { lng } = (req || {}).query || {};
+	if (lng) {
+		return lng;
+	}
+
+	// get language by cookie
+	if (req && req.cookies && req.cookies.USER_LANG) {
```
Comment (mwolowyk-cap) [inline helpers/i18n.js:53]: please check whether it is necessary

File: middleware/i18n.js [hunk 2]
```diff
@@ -1,8 +1,13 @@
-const { getInstance, getUserLanguage } = require('../helpers/i18n');
+const { getInstance, changeLanguage, getCurrentLanguage } = require('../helpers/i18n');
 
 const middleware = (req, res, next) => {
-	res.locals.userLanguage = getUserLanguage(res.locals.currentUser);
-	res.$t = getInstance(res.locals.currentUser);
+	const currentLanguage = getCurrentLanguage(req, res);
+	if (currentLanguage) {
+		changeLanguage(currentLanguage);
+		res.cookie('USER_LANG', currentLanguage);
```
Comment (mwolowyk-cap) [inline middleware/i18n.js:7]: please check whether it is necessary

**Review by mwolowyk-cap — COMMENTED** [review]
@DanielRieckhof please see the comments

File: helpers/i18n.js (hunk 1 above)
Comment (mwolowyk-cap) [reply helpers/i18n.js:47]: yes necessary :)

File: config/default.schema.json [hunk 3]
```diff
@@ -317,6 +317,32 @@
...
+				},
+				"FALLBACK_LANGUAGE": {
+					"type": "string",
+					"default": "de",
+					"description": "Value for the fallback language if translation key is not found"
+				},
+				"DEBUG": {
+					"type": "boolean",
+					"default": false,
+					"description": "Enable logging"
```
Comment (mwolowyk-cap) [inline config/default.schema.json:342]: please clarify it.

File: helpers/i18n.js [hunk 4]
```diff
@@ -2,21 +2,26 @@ const i18next = require('i18next');
...
+const { Configuration } = require('@schul-cloud/commons');
 const logger = require('./logger');
 
-const defaultLanguage = 'de';
-const localeDir = path.join(__dirname, '../locales');
+const i18nDebug = Configuration.get('I18N__DEBUG');
+const i18nFallbackLanguage = Configuration.get('I18N__FALLBACK_LANGUAGE');
+const i18nDefaultLanguage = Configuration.get('I18N__DEFAULT_LANGUAGE');
 
+const localeDir = path.join(__dirname, '../locales');
```
Comment (mwolowyk-cap) [inline helpers/i18n.js:12]: please add caching layer for the available language

File: middleware/i18n.js (hunk 2 above)
Comment (mwolowyk-cap) [reply middleware/i18n.js:7]: yes it is necessary )

**Review by mwolowyk-cap — COMMENTED** [review]
see code comments

**Review by Metauriel — APPROVED** [review]

File: config/default.json [hunk 5]
```diff
@@ -9,5 +9,11 @@
 	"COOKIE_SECRET": "ihasDF(=4jA4aF",
 	"STUDENT_TEAM_CREATION": "opt-out",
 	"FEATURE_EXTENSION_PORTFOLIO_ENABLED": false,
-	"LERNSTORE_MODE": "DISABLED"
+	"LERNSTORE_MODE": "DISABLED",
+	"I18N": {
+		"DEFAULT_LANGUAGE": "de",
+		"FALLBACK_LANGUAGE": "de",
+		"DEBUG": false,
+		"AVAILABLE_LANUAGES": "de,en"
```
Comment (Tratori) [inline config/default.json:17]: small typo here 
```suggestion
		"AVAILABLE_LANGUAGES": "de,en"
```

File: locales/de.json [hunk 6]
```diff
@@ -2205,6 +2211,12 @@
 				"theNextQuestionnaireIsOnline": "Die nächste Befragung zum Projekt {{shortTitle}} ist online. Gerne möchten wir Sie erneut um Ihre Mitarbeit bitten! Die Bearbeitung wird ca. 20 - 25 Minuten Ihrer Zeit in Anspruch nehmen."
 			}
 		},
+		"form_ChangeLanguage" : {
+			"label": "Sprache",
```
Comment (Tratori) [inline locales/de.json:2215]: Keys should follow this schema: `$Location.$Type.$Description`
```suggestion
			"label": {
				"language" : "Sprache"
			},
```

File: locales/en.json [hunk 7]
```diff
@@ -37,18 +37,59 @@
...
 		}
 	},
 	"homework": {
 		"button": {
 			"downloadUngradedFiles": "Download files of ungraded homework"
 		}
 	},
+	"lib": {
+		"form_ChangeLanguage" : {
+			"label": "Language",
```
Comment (Tratori) [inline locales/en.json:55]: (same here)

Comment (ABialoskorski) [reply locales/en.json:55]: image
needs to change

File: config/default.json (hunk 5 above)
Comment (ABialoskorski) [reply config/default.json:17]: done ;)

File: locales/de.json (hunk 6 above)
Comment (ABialoskorski) [reply locales/de.json:2215]: done ;)

File: locales/en.json (hunk 7 above)
Comment (ABialoskorski) [reply locales/en.json:55]: done ;)

File: helpers/i18n.js (hunk 4 above)
Comment (DanielRieckhof) [reply helpers/i18n.js:15]: Moved auto-detection to config

**Review by monaweitzenberg — APPROVED** [review]

**Pull request merged**